src/components/App/index.tsx

```diff
@@ -59,23 +60,28 @@ const filesize = partial({});
 
 async function compressImage(
   source: SourceImage,
+  quantizerOptions: quantizer.QuantizeOptions | null,
   encodeData: EncoderState,
 ): Promise<File> {
   // Special case for identity
   if (encodeData.type === identity.type) return source.file;
 
+  let sourceData = source.data;
+  if (quantizerOptions) {
+    sourceData = await quantizer.quantize(sourceData, quantizerOptions);
```

Comment:
Thought: would it be easier to mix the quantizer options into EncoderState and then peel them off here?

Reply:
In terms of naming that would make sense. I might give that version a spin tomorrow.